tests/foreman/cli/test_host_collection.py

```diff
@@ -363,3 +359,43 @@ def test_positive_list_hosts(self):
             u'organization-id': self.org['id']
         })
         self.assertEqual(new_system['name'].lower(), result[0]['name'])
+
+    @tier2
+    def test_positive_host_collection_host_pagination(self):
+        """Check if pagination configured on per-page param defined in hammer
+        host-collection hosts command overrides global configuration defined
+        on /etc/hammer/cli_config.yml, which default is 20 per page
+
+        @BZ: 1343583
+
+        @id: bbe1108b-bfb2-4a03-94ef-8fd1b5a0ec82
+
+        @Assert: Number of host per page follows per_page configuration
+        restriction
+
+        @CaseLevel: Integration
+        """
+        host_collection = self._new_host_collection({
+            'name': gen_string('alpha', 15)
+        })
+        host_ids = ','.join(
+            self._make_content_host_helper()['id'] for i in range(2)
+        )
+        HostCollection.add_host({
+            u'host-ids': host_ids,
+            u'id': host_collection['id'],
+            u'organization-id': self.org['id'],
+        })
+
+        def assert_hosts_per_page(n):
```

Comment:
Also it can be extended easily using range function or maybe creating a list of random numbers... But this is not the scope of this test.